### ISSUE TITLE
False positive for Style/ReturnNilInPredicateMethodDefinition cop

### ISSUE THREAD

**Issue by brunomrlima** [rubocop/rubocop#12073]
## Expected behavior

`rubocop` should not detect any issues.

## Actual behavior

`rubocop` is detecting a `Style/ReturnNilInPredicateMethodDefinition` issue.

## Steps to reproduce the problem

The code below returns a boolean, but `rubocop` thinks it returns `nil`.

```ruby
[1, 2, 3].all?(nil)
```

## RuboCop version

```
$ [bundle exec] rubocop -V
1.55.0 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.0.6) [x86_64-darwin22]
  - rubocop-performance 1.18.0
  - rubocop-rails 2.20.2
  - rubocop-rspec 2.22.0
```

**Comment by koic** [comment]
This issue is a dup with #12068.
